torch/_dynamo/variables/builtin.py

```diff
@@ -1881,6 +1896,16 @@ def call_dict(self, tx: "InstructionTranslator", *args, **kwargs):
 
     @staticmethod
     def call_custom_dict(tx: "InstructionTranslator", user_cls, *args, **kwargs):
+        args = list(args)
+        if (
+            len(args) == 1
+            and isinstance(args[0], variables.GetAttrVariable)
+            and not tx.output.side_effects.has_pending_mutation(args[0].obj)
```

Comment:
How do we ensure writes to `obj` and the dict are mirrored _after_ the construction here?